backend/playerManager.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * Handles the state of each turn and manages the players.
+ * @param players list of Player objects in the game
+ */
+class PlayerManager {
+    constructor(players) {
+        this.turnOrder = players;
+
+        this.turnIndex = 0;
+        this.currentPlayer = players[0];
+        this.canRoll = true; // says if the current player can roll
+
+        this.doubleCount = 0;
+    }
+
+    /**
+     * Chooses the order of the players for the game.
+     */
+    scrambleTurnOrder() {
+        for (let i = this.turnOrder.length; i; i--) {
+            let j = Math.floor(Math.random() * i);
+            [this.turnOrder[i - 1], this.turnOrder[j]] = [this.turnOrder[j], this.turnOrder[i - 1]];
+        }
+    }
+
+    /**
+     * Makes it so the next player in the order goes.
+     */
+    nextTurn() {
+        this.turnIndex += 1
```

Comment:
this makes sure that there will never be overflow issues 👍 (also do `:+1` instead of +1) lol